lib/reader.ts

```diff
@@ -327,13 +337,20 @@ class ParquetReader {
  * rows from a parquet file use the ParquetReader instead
  */
 let ParquetEnvelopeReaderIdCounter = 0;
-class ParquetEnvelopeReader {
-
-  static async openFile(filePath, options) {
+export class ParquetEnvelopeReader {
+  readFn: Function;
+  close: Function;
+  id: number;
+  fileSize: Function | number;
```

Comment:
I don't see anywhere that `fileSize` is invoked, but likely just missing it. This is a dangerous type. Assuming it is correct, we should likely add an issue to fix it.

Reply:
Yea I first imagined it to be a number but noticed that an async function is being passed as `filesize` on line 408 which is what prompted me to leave as `Function`

Reply:
I think that is a different variable, but not 100% sure. `fileSize` vs `filesize` is not good.